cartoframes/data/services/isolines.py

```diff
@@ -148,11 +156,14 @@ def _iso_areas(self,
                 raise Exception('No valid geometry found. Please provide an input source with ' +
                                 'a valid geometry or specify the "geom_col" param with a geometry column.')
 
-            to_carto(source_cdf, temporary_table_name, self._credentials, log_enabled=False)
+            index_as_cartodbid = CARTO_INDEX_KEY not in source_cdf.columns
+
+            to_carto(source_cdf, temporary_table_name, self._credentials, index=index_as_cartodbid,
+                     index_col=CARTO_INDEX_KEY, log_enabled=False)
             source_query = 'SELECT * FROM {table}'.format(table=temporary_table_name)
 
-        source_columns = source_manager.get_column_names()
-        source_has_id = 'cartodb_id' in source_columns
+        if source_col is None:
+            source_col = CARTO_INDEX_KEY
 
         iso_function = '_cdb_{function}_exception_safe'.format(function=function)
         # TODO: use **options argument?
```

Comment:
⚠️ random TODO with no issue referenced. I'd 🔥 or create an issue explaining it so it can be prioritized at some point. (probably 🔥 since we don't know what is this about...)